src/__test__/simulator/index.ts

```diff
@@ -1,53 +1,82 @@
 import _sodium from 'libsodium-wrappers';
-import { removeUser, signInFirebase } from './firebase';
-import { sdk } from './sdk';
-
-export const signIn = async () => {
-	try {
-		await _sodium.ready;
-		const data = await signInFirebase();
-		sdk.setUid(data.user.uid);
-	} catch (error) {
-		console.error("signIn sim err", error);
+import { Firebase } from './firebase';
+import { SimulatorSdk } from './sdk';
+import { v4 as uuid } from 'uuid';
+
+export class Simulator {
+	sdk: SimulatorSdk | null = null;
+	firebase: Firebase;
+
+	constructor() {
+		this.firebase = Firebase.instance();
 	}
-};
-
-export const cleanUpSimulation = async () => {
-	try {
-		await removeUser();
-		sdk.cleanState();
-		console.log("cleanUpSimulation sim done");
-	} catch (error) {
-		console.error("cleanUpSimulation sim err", error);
+
+	static async init() {
+		await SimulatorSdk.init();
 	}
-};
 
-export const pairing = async (qrCode: string, isRepair = false) => {
-	try {
-		await sdk.sendPairing(qrCode, isRepair);
-	} catch (error) {
-		console.error("pairing sim err", error);
+	getSdk = () => {
+		if (this.sdk == null) {
+			throw new Error('Do signin before using sdk');
+		}
+		return this.sdk;
 	}
-};
 
-export const keygen = async () => {
-	try {
-		await sdk.keygen();
-	} catch (error) {
-		console.error("keygen sim err", error);
+	signIn = async () => {
+		try {
+			await this.firebase.signInFirebase();
+			// Using random UUID for testing only.
+			this.sdk = new SimulatorSdk(uuid(), this.firebase.db);
```

Comment:
We need to hardcode the "id" of simulator, which is mentioned in our previous discussion to reduce scope of exposure on staging Firestore